pkg/operator/ceph/pool/health.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pool
+
+import (
+	"time"
+
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/clusterd"
+	cephclient "github.com/rook/rook/pkg/daemon/ceph/client"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	defaultHealthCheckInterval = 1 * time.Minute
+)
+
+type mirrorChecker struct {
+	context         *clusterd.Context
+	interval        time.Duration
+	client          client.Client
+	clusterInfo     *cephclient.ClusterInfo
+	namespacedName  types.NamespacedName
+	healthCheckSpec *cephv1.MirrorHealthCheckSpec
+	poolName        string
+}
+
+// newMirrorChecker creates a new HealthChecker object
+func newMirrorChecker(context *clusterd.Context, client client.Client, clusterInfo *cephclient.ClusterInfo, namespacedName types.NamespacedName, healthCheckSpec *cephv1.MirrorHealthCheckSpec, poolName string) *mirrorChecker {
+	c := &mirrorChecker{
+		context:         context,
+		interval:        defaultHealthCheckInterval,
+		clusterInfo:     clusterInfo,
+		namespacedName:  namespacedName,
+		client:          client,
+		healthCheckSpec: healthCheckSpec,
+		poolName:        poolName,
+	}
+
+	// allow overriding the check interval
+	checkInterval := healthCheckSpec.Mirror.Interval
+	if checkInterval != "" {
+		if duration, err := time.ParseDuration(checkInterval); err == nil {
```

Comment:
What if `err != nil`? Would the interval be 0 and the `checkMirroring()` loop would keep checking in a tight loop?

Reply:
No, I believe it won't get set and will have `defaultHealthCheckInterval` as a default value. See L47.